### PR TITLE
[WFLY-16677] Turn off builds of legacy javax.* production modules; re…

### PR DESCRIPTION
**Pull request by bstansberry**
…move dependencies upon them.

Clean out additional boms/legacy-xx dependencyManagement entries for items no longer used once these builds are disabled.

Note that this leaves likely unused legacy.version.xx properties behind in the root pom; cleaning those up would be follow-up work.

Note also that completion of #15863 will allow for removal of further legacy entries.

https://issues.redhat.com/browse/WFLY-16677

### CODE REVIEW

File: clustering/infinispan/extension/pom.xml [hunk 1]
```diff
@@ -86,10 +104,6 @@
             <groupId>${project.groupId}</groupId>
             <artifactId>wildfly-clustering-server-service</artifactId>
         </dependency>
-        <dependency>
-            <groupId>${project.groupId}</groupId>
-            <artifactId>wildfly-transactions</artifactId>
-        </dependency>
         <dependency>
             <groupId>org.wildfly.transaction</groupId>
             <artifactId>wildfly-transaction-client</artifactId>
```
Comment (yersan) [inline clustering/infinispan/extension/pom.xml:109]: There is a -jakarta variant for wildfly-transaction-client, I guess we could use it here instead.

**Review by yersan — APPROVED** [review]
Independent of my previous comment, giving a symbolic approval. We can refine any pending  dependencies later, all CI are in green, so it should not prevent merging this

File: clustering/infinispan/extension/pom.xml (hunk 1 above)
Comment (bstansberry) [reply clustering/infinispan/extension/pom.xml:109]: @yersan Thanks. I picked that up in my v2 version of this -- #15871.

**Comment by bstansberry** [comment]
Thanks for the review, @yersan. I'm going to close this one as there are now 3 further variants derived from this, all of which would be used instead of this one.

**Pull request closed without merging**